### PR TITLE
Feature: Add REST API skeleton

### LINKED ISSUE #99: Generate Flask code from postman
## User stories
As a developer I want to work with the REST API in order to evaluate and test my work. 

## Dependencies
Depends on: 
 - #52 
 - #141 

## Acceptance Criteria
- [x] Generate code from flask app (contains the implementation of the components REST API and Implementor)
- [x] Add poe commands with loading env files

### PR DESCRIPTION
**Pull request by antonykamp**
Fixes #99

This PR adds a skeleton of the REST API, including empty implementors and tests.
It installs flask and wegargs and adds poe commands for starting the development server.

During development I recognized some mistakes:
- The naming schema of the operationIds when fetching all ids is not uniform (see #262)
- The endpoint for adding, editing, and deleting PlatFormBlockedFaults is missing (see #263)
- Typo "schedule_blockedfault" (see #264)
- Typo "TrackSplitLimitFault" (see #267)  
- We use the wrong return types for a successful PUT operation (see #268)

Currently, the implementation uses marshmallow schemas from `src/schemas/model.py`.
When we implement the implementors, we'll replace the used schemas with the real ones.
At a later point, we'll remove the file `src/schemas/model.py` when it's not needed anymore (see #269) 

## PR checklist

- [x] Acceptance criteria fulfilled
- [x] Additional features are tested
- [x] Docs (code, wiki & diagrams) updated
- [ ] Dev-branch has been merged into local branch to resolve conflicts
- [ ] Tests and linter have passed AFTER local merge
- [ ] Another dev reviewed and approved

### CODE REVIEW

File: src/api/component.py [hunk 1]
```diff
@@ -0,0 +1,380 @@
...
+    """Create a schedule blocked fault configuration"""
+    schema = model.ScheduleBlockedFaultConfiguration()
+
+    body = parser.parse(schema, request, location="json")
+
+    return impl.component.create_schedule_blocked_fault_configuration(body)
+
+
+@bp.route(
+    "/component/fault-injection/schedule-blockedfault/<identifier>", methods=["get"]
```
Comment (antonykamp) [inline src/api/component.py:30]: Typo will be fixed in #264

File: src/api/component.py [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+    """Create a track speed limit fault configuration"""
+    schema = model.TrackSpeedLimitFaultConfiguration()
+
+    body = parser.parse(schema, request, location="json")
+
+    return impl.component.create_track_speed_limit_fault_configuration(body)
+
+
+@bp.route(
+    "/component/fault-injection/track-split-limit-fault/<identifier>", methods=["get"]
```
Comment (antonykamp) [inline src/api/component.py:142]: Typo will be fixed in #267

File: src/implementor/schedule.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    :param options: A dictionary containing all the paramters for the Operations
+        options["id"]
+
+    :param body: The parsed body of the request
+    """
+
+    # Implement your business logic here
+    # All the parameters are present in the options argument
+
+    return "", 204
```
Comment (antonykamp) [inline src/implementor/schedule.py]: Wrong status code will be fixed in #268

File: src/implementor/run.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
+# pylint: disable=unused-argument
+# pylint: disable=duplicate-code
+
+import json
+
+
+def get_all_run_id(options):
```
Comment (antonykamp) [inline src/implementor/run.py:7]: Missing uniformly naming will be fixed in #262

File: src/implementor/run.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    """
+    :param options: A dictionary containing all the paramters for the Operations
+        options["simulationId"]
+
+    """
+
+    # Implement your business logic here
```
Comment (antonykamp) [inline src/implementor/run.py:14]: **QUESTION:** Should we throw `NotImplementedError`s here?

**Review by instant-sky — APPROVED** [review]
Very nice work! Thank you for explaining this PR, that helped a lot!

**Pull request merged**